tests/test_basic_ops_list.py

```diff
@@ -845,6 +845,108 @@ def test_getitem_slice(sli: slice) -> None:
     assert olst.__getitem__(sli) == expected
 
 
+@mark.parametrize(
+    "constructor",
+    [OmegaConf.create, list, lambda lst: OmegaConf.create({"foo": lst}).foo],
+)
+@mark.parametrize(
+    "lst, idx, value, expected",
+    [
+        param(
+            ["a", "b", "c", "d"],
+            slice(1, 3),
+            ["x", "y"],
+            ["a", "x", "y", "d"],
+            id="same-number-of-elements",
+        ),
+        param(
+            ["a", "x", "y", "d"],
+            slice(1, 3),
+            ["x", "y", "z"],
+            ["a", "x", "y", "z", "d"],
+            id="extra-elements",
+        ),
+        param(
+            ["a", "x", "y", "z", "d"],
+            slice(1, 1),
+            ["b"],
+            ["a", "b", "x", "y", "z", "d"],
+            id="insert only",
+        ),
+        param(
+            ["a", "b", "x", "y", "z", "d"],
+            slice(1, 1),
+            [],
+            ["a", "b", "x", "y", "z", "d"],
+            id="nop",
+        ),
+        param(
+            ["a", "b", "x", "y", "z", "d"],
+            slice(1, 3),
+            [],
+            ["a", "y", "z", "d"],
+            id="less-elements",
+        ),
+        param(
+            ["a", "y", "z", "d"],
+            slice(1, 2, 1),
+            ["b"],
+            ["a", "b", "z", "d"],
+            id="extended-slice",
+        ),
+        param(
+            ["a", "b", "c", "d"],
+            slice(1, 3, 1),
+            ["x", "y"],
+            ["a", "x", "y", "d"],
+            id="extended-slice2",
+        ),
+        param(
+            ["a", "b", "z", "d"],
+            slice(0, 3, 2),
+            ["a", "c"],
+            ["a", "b", "c", "d"],
+            id="extended-slice-disjoint",
+        ),
+        param(
+            ["a", "b", "c", "d"],
+            slice(1, 3),
+            1,
+            raises(TypeError),
+            id="non-iterable-input",
+        ),
+        param(
+            ["a", "b", "c", "d"],
+            slice(1, 3, 1),
+            ["x", "y", "z"],
+            ["a", "x", "y", "z", "d"],
+            id="extended-slice-length-mismatch",
+        ),
+        param(
+            ["a", "b", "c", "d", "e", "f"],
+            slice(1, 5, 2),
+            ["x", "y", "z"],
+            raises(ValueError),
+            id="extended-slice-length-mismatch2",
+        ),
+    ],
+)
+def test_setitem_slice(
+    lst: List[Any],
+    idx: slice,
+    value: Union[List[Any], Any],
+    expected: Union[List[Any], RaisesContext],
```

Comment:
```suggestion
    expected: Union[List[Any], RaisesContext[Any]],
```
^ Hopefully this will fix the mypy error. Sorry about that.